### DISCUSSION TITLE
Extension placement fix / moving the pagination

### DISCUSSION THREAD

**Discussion by ApfailKuchen** [templaza/astroid-framework#642] [Q&A] [answered]
Hello everyone,

Firstly, I want to apologize and hope you can understand me. English isn't my native language.

I'm encountering an issue with an extension: I bought an "Item Review" extension from Joomunited for managing reviews.
[Yes, Astroid has a built-in feature for this, but it's not as advanced.]

The problem is this: 
The extension is currently positioned below the article content, which is correct. However, the page navigation is situated right beside it rather than at the bottom, where it should be the last item. 

[Screenshot](https://ibb.co/vQnhbx7)

But how can this be changed?

My thought is there must be a way for Astroid to determine where to put the code from this extension.

After a short search I found a couple of approaches in /templates/tp_surfing/html/com_content/article/default.php:

onContentAfterTitle
onContentBeforeDisplay
onContentAfterDisplay

The developer of the extension provided me with this guidance:

"We've used the content plugin code, so the code you need to modify is located in the article template file:"

`components\com_content\tmpl\article\default.php`

```
  <?php echo $this->item->event->afterDisplayContent; ?>
Our entire code is contained within the following file:
\components\com_itemrating\helpers\itemrating.php
```

Now, I'm uncertain if changing everything to onContentAfterDisplay instead of afterDisplayContent will solve the issue, or if it's even the correct approach.

That's why I'm reaching out to those with more expertise in this area, hoping you might be able to assist me.

**Comment by Chacapamac** [comment]
> "We've used the content plugin code, so the code you need to modify is located in the article template file:"
> 
> components\com_content\tmpl\article\default.php
> 
**This is the way You can  tailor Joomla and good 3rd party extensions** — Just make an override (in the html folder of your template)

Find the container of the code displaying these buttons and move the block after the content.
Ideally, if you not a big php, html coder, test on a copy of the website.

**Reply by ApfailKuchen** [reply]
Hey @Chacapamac thanks for your answer. 

Please excuse me for only replying today. I keep having to go to the hospital because of my cancer treatment.

i have change the code from the file `components\com_content\tmpl\article\default.php`
I moved the code parts but unfortunately it had no effect


New
```
    <?php // Content is generated by content plugin event "onContentAfterDisplay" ?>
    <?php echo $this->item->event->afterDisplayContent; ?>
    <?php
    if (!empty($this->item->pagination) && $this->item->paginationposition && $this->item->paginationrelative) :
        echo $this->item->pagination;
        ?>
    <?php endif; ?>
```

The page navigation remains at the height of the extension and does not go below it.

URL: https://clone.apfailkuchen.de
Example: https://clone.apfailkuchen.de/android/test

---------

Do you perhaps have an idea how to move the page navigation down?

**Reply by ApfailKuchen** [reply]
I now have the correct file, which also affects the linked page, but unfortunately I don't know how to make the change.
[default.txt](https://github.com/templaza/astroid-framework/files/14980524/default.txt)

I started at the end of the PHP code, where the renderAuthorInfo is mentioned.

I then put the page navigation above there.

But unfortunately this had no effect.

The piece of code for page navigation that is needed here is in line 153, but how do I get this under the renderAuthorInfo?

**Comment by ApfailKuchen** [comment, marked as answer]
I was able to solve the problem. I made the following code changes:

**Old**
```
	<?php     
      if (!empty($this->item->pagination) && $this->item->pagination && $this->item->paginationposition && !$this->item->paginationrelative) :
         echo $this->item->pagination;
      ?>
      <?php endif; ?>
```


**New**
```
<?php

function renderPaginationIfNeeded($item) {
    if (!empty($item->pagination) && $item->pagination && $item->paginationposition && !$item->paginationrelative) {
        echo $item->pagination;
    }
}

?>
```

and then added the code under  renderAuthorInfo: 

 ```
  <?php $astroidArticle->renderSocialShare(); ?>
   <?php $astroidArticle->renderAuthorInfo(); ?>
   <?php renderPaginationIfNeeded($this->item); ?>
   <?php $astroidArticle->renderComments(); ?>
   <?php $astroidArticle->renderRelatedPosts(); ?>
```
